File: python/word-count/wordcount.py
import re

def word_count(sentence):
    split_sentence = re.split(' +| *\n *|\t', sentence)
    output = {}
    for x in split_sentence:
        output[x] = count_word(x, split_sentence)
    return output

def count_word(word, iterable):
    count = 0
    for x in iterable:
        if x == word:
            count += 1
    return count

File: elixir/word-count/word_count.exs
defmodule Words do
  @doc """
  Count the number of words in the sentence.

  Words are compared case-insensitively.
  """
  @spec count(String.t) :: map()
  def count(sentence) do
    sentence |> split_sentence |> convert_to_words_and_count |> Enum.into(%{})
  end

  def split_sentence(sentence) do
    sentence |> format_and_filter |> String.split
  end

  def format_and_filter(sentence) do
    sentence |> String.replace(~r/[^\w -]+/u, "") |> String.replace(~r/_/, " ") |> String.downcase
  end

  def convert_to_words_and_count(words) do
    Enum.map(words, fn y -> {y, Enum.count(words, fn z -> z == y end)} end)
  end
end
